sdk/search/azure-search-documents/src/test/java/com/azure/search/documents/test/environment/setup/AzureSearchResources.java

```diff
@@ -114,13 +115,21 @@ public void createService(TestResourceNamer testResourceNamer) {
         System.out.println("Creating Azure Cognitive Search service: " + searchServiceName);
         int recreateCount = 0;
         do {
-            searchService = azure.searchServices()
-                .define(searchServiceName)
-                .withRegion(location)
-                .withExistingResourceGroup(resourceGroup)
-                .withFreeSku()
-                .create();
-            recreateCount += 1;
+            try {
+                searchService = azure.searchServices()
+                    .define(searchServiceName)
+                    .withRegion(location)
+                    .withExistingResourceGroup(resourceGroup)
+                    .withFreeSku()
+                    .create();
```

Comment:
If create succeeded, then you can break out of the `do-while` loop.

Reply:
The problem is the management plan does not provide us the status of create. I have to ping the service to confirm. The newly added code is to avoid failure of already exists.

Reply:
Could we check the `status` of the `SearchService` object?

Reply:
Just checked over the API. Find the status one this time. Missing from the initial investigate.
Will try to switch to status() checking

Reply:
I have tried the API they provided. Seems not that stable as service ping. In order to make our live tests more stable. I decided to switch back to ping machenism.

Reply:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=356026&view=results